logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java

```diff
@@ -176,8 +178,11 @@ public HttpHeaders getHeaders() {
 
     @Override
     public Charset getCharset() {
-        return Optional.ofNullable(getCharacterEncoding())
-                .map(Charset::forName)
+        String contentTypeHeader = getHeaders().getFirst(CONTENT_TYPE_HEADER);
```

Comment:
I'm unsure whether this should be `getFirst` or `getLast`. I tested sending multiple content-type headers to a server using `curl`, and it seemed that the server used the last one.

Reply:
Good question. I couldn't find any recommendations in this regard in the RFC, I think multiple header values overall should not be expected for Content-Type header. Here I followed the same approach we had in other classes (everywhere `getFirst()` is used). Bur if you find any leads about recommendations here, please share. I was mostly going through https://www.rfc-editor.org/rfc/rfc9110.html#fields

Reply:
I also couldn't find a reference on this. Yesterday, I read that only comma-separated headers (arrays for that matter) can be repeated multiple times. To test this non-standard behavior, I created a simple Python application (Flask) that echoed the request headers in response body, and sent the following curl command:

```bash
curl -v http://127.0.0.1:8080/headers -H 'content-type: application/json' -H 'content-type: text'
```

It returned the **last** `content-type`. Today, I tested it with the following Spring Boot Web controller:

```java
@RestController
@RequestMapping("/headers")
public class HeadersController {

    @GetMapping
    public String getRequestHeaders(@RequestHeader Map<String, String> headers) {
        StringBuilder sb = new StringBuilder();
        headers.forEach((key, value) -> sb.append(key).append(": ").append(value).append("\n"));
        return sb.toString();
    }
}
```

It returned the **first**  `content-type`. We can conclude that there is no standard behavior, and therefore the code is OK 👌 .

Reply:
Thank you for verifying! I'll resolve the thread then.